### PR TITLE
Mission Decodable

### PR DESCRIPTION
**Pull request by jbethanco**
### Discussion :speech_balloon: 
It has been requested that we can download previously uploaded missions. In order to do this, we need to make all our NSManagedObjects decodable in addition to encodable. 

### What this PR does :thinking: 
Adds tons of boilerplate for Decodable Conformance to All Entities
Remove Encodable Files
Move Encodable Functions to Classes
Add Decodable Tests
Remove FlightTime.totalTime from data model as we only calculate it
Remove Person.organization as it is not used. We use crewLine.flyingOrganization
Remove the same from the JSON generation tests.

### Resolves :white_check_mark: 
https://jira.il2.dso.mil/browse/PBS-1823 Mission Decodable

### Testing Items :microscope: 
- [x] Verify the tests are robust and verifying what we need them to test.

### Notes :writing_hand: 
Unable to override isEqual on NSManagedObject which is why the test is so verbose. 

### UI Changes :eye: 
None.

### Resources :books: 
https://www.donnywals.com/using-codable-with-core-data-and-nsmanagedobject/

### CODE REVIEW

File: LoggingTests/JSONDecodingTest.swift [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
...,"other":0.0,"primary":0.0,"secondary":0.0},"flightConditions":{"combatSupportTime":null,"combatTime":null,"instruments":0.0,"night":0.0,"nvg":0.0,"simInstruments":0.0},"missionEventRecords":[]}],"fuel":{"airRefuel":0.0,"land":0.0,"ramp":0.0},"metrics":{"airdropWeight":0.0,"airlandWeight":0.0,"numAirdropPallets":0,"numAirlandPallets":0,"numHeavies":0,"numJumpers":0,"numPassengers":0,"numRollingStock":0},"sortieType":{"isAeromedevac":false,"isAirdrop":false,"isAirland":false,"isAirRefueling":false,"isAMCForm54Filed":false,"isAMCForm97Filed":false,"isAskTACCFiled":false,"isCombat":false,"isCombatSupport":false,"isDistinguishedVisitor":false,"isHumanitarian":false,"isSortieDelay":false,"isSpecialOpsLL2":false,"sortieDelayRemarks":""},"form781Remarks":""}],"additionalMissionInformation":{"timeApprovedByADO":null,"timeApprovedByMx":null,"timeApprovedBySARM":null,"mxApprovedBy":null,"adoApprovedBy":null,"sarmApprovedBy":null,"timeReceived":"2021-09-09T23:03:40.087161Z","timeRevised":null,"createdBy":"eff3c95f-b68c-4a7c-b95e-fbb0a07b6f71","lastModifiedBy":"eff3c95f-b68c-4a7c-b95e-fbb0a07b6f71","createdDate":"2021-09-09T23:03:40.097315Z","lastModifiedDate":"2021-09-09T23:03:40.097315Z"}}"#
```
Comment (mcgarrys) [inline LoggingTests/JSONDecodingTest.swift:150]: Add the swiftlint ignore next line for these two declarations

File: LoggingTests/JSONDecodingTest.swift [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+                XCTAssertEqual(testSortie.takeoffTime, originalSortie.takeoffTime)
+                XCTAssertEqual(testSortie.landTime, originalSortie.landTime)
+                XCTAssertEqual(testSortie.missionNumber, originalSortie.missionNumber)
+                XCTAssertEqual(testSortie.serialNumber, originalSortie.serialNumber)
+                XCTAssertEqual(testSortie.flightAuthNumber, originalSortie.flightAuthNumber)
+                XCTAssertEqual(testSortie.missionSymbol, originalSortie.missionSymbol)
+                XCTAssertEqual(testSortie.missionNumber, originalSortie.missionNumber)
+                XCTAssertEqual(testSortie.pilotReviewInitials, originalSortie.pilotReviewInitials)
+                XCTAssertEqual(testSortie.unitCharged, originalSortie.unitCharged)
+                XCTAssertEqual(testSortie.specialUse, originalSortie.specialUse)
```
Comment (mcgarrys) [inline LoggingTests/JSONDecodingTest.swift:64]: Missing an assertion for testSortie.mds

**Review by mcgarrys — COMMENTED** [review]
Looks like a few sortie properties were missed in the JSONDecodingTest. Also, do we not need to conform Form791/Issues to Codable?

File: LoggingTests/JSONDecodingTest.swift (hunk 2 above)
Comment (mcgarrys) [reply LoggingTests/JSONDecodingTest.swift:64]: Sorry, not sure why my comment didn't update. Looks like mds, issuingUnit, harmLocation, numTouchAndGoes, and numFullStops are missing (unless you left them out intentionally?).

Comment (jbethanco) [reply LoggingTests/JSONDecodingTest.swift:64]: Good catches.

**Review by mcgarrys — APPROVED** [review]
Looks good, test passes locally 🧠

File: PBLogging/Data/EntityClassCodable/Mission+CoreDataClass.swift [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+@objc(Mission)
+public class Mission: NSManagedObject, Codable {
+    
+    //    static func decodeMissionData(_ mission: Data, context: NSManagedObjectContext) throws -> Mission {
+    //        let decoder = JSONDecoder()
+    //        decoder.userInfo[CodingUserInfoKey.managedObjectContext] = context
+    //
+    //        return Mission()
+    //    }
```
Comment (bmorris-revacomm) [inline PBLogging/Data/EntityClassCodable/Mission+CoreDataClass.swift:20]: delete if not using?

Comment (jbethanco) [reply PBLogging/Data/EntityClassCodable/Mission+CoreDataClass.swift]: Nice catch

**Review by bmorris-revacomm — APPROVED** [review]
🔦

**Pull request merged**